src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Parser/SbomPackageParser.cs

```diff
@@ -0,0 +1,355 @@
+    private const string LicenseConcludedProperty = "licenseConcluded";
+    private const string LicenseDeclaredProperty = "licenseDeclared";
+    private const string CopyrightTextProperty = "copyrightText";
+    private const string VersionInfoProperty = "versionInfo";
+    private const string SupplierProperty = "supplier";
+    private const string PackageVerificationCodeProperty = "packageVerificationCode";
+    private const string PackageVerificationCodeValueProperty = "packageVerificationCodeValue";
+    private const string PackageVerificationCodeExcludedFilesProperty = "packageVerificationCodeExcludedFiles";
+    private const string LicenseInfoFromFilesProperty = "licenseInfoFromFiles";
+    private const string HasFilesProperty = "hasFiles";
+    private const string ExternalRefsProperty = "externalRefs";
+    private readonly Stream stream;
+    private readonly SPDXPackage sbomPackage = new ();
+
+    public SbomPackageParser(Stream stream)
+    {
+        this.stream = stream ?? throw new ArgumentNullException(nameof(stream));
+    }
+
+    internal long GetSbomPackage(ref byte[] buffer, ref Utf8JsonReader reader, out SPDXPackage sbomPackage)
+    {
+        if (buffer is null || buffer.Length == 0)
+        {
+            throw new ArgumentException($"The {nameof(buffer)} value can't be null or of 0 length.");
+        }
+
+        try
+        {
+            // If the end of the array is reached, return with null value to signal end of the array.
+            if (reader.TokenType == JsonTokenType.EndArray)
+            {
+                sbomPackage = null;
+                return 0;
+            }
+
+            // Read the start { of this object.
+            ParserUtils.SkipNoneTokens(stream, ref buffer, ref reader);
+            ParserUtils.AssertTokenType(stream, ref reader, JsonTokenType.StartObject);
+
+            // Move to the first property name token.
+            ParserUtils.Read(stream, ref buffer, ref reader);
+            ParserUtils.AssertTokenType(stream, ref reader, JsonTokenType.PropertyName);
+
+            while (reader.TokenType != JsonTokenType.EndObject)
+            {
+                ParseProperty(ref reader, ref buffer);
+
+                // Read the end } of this object or the next property name.
+                ParserUtils.Read(stream, ref buffer, ref reader);
+            }
+
+            // Validate the created object
+            ValidateSbomPackage(this.sbomPackage);
+
+            sbomPackage = this.sbomPackage;
+            return reader.BytesConsumed;
+        }
+        catch (EndOfStreamException)
+        {
+            sbomPackage = null;
+            return 0;
+        }
+        catch (JsonException e)
+        {
+            sbomPackage = null;
+            throw new ParserException($"Error while parsing JSON, addtional details: ${e.Message}", e);
+        }
+    }
+
+    private void ValidateSbomPackage(SPDXPackage sbomPackage)
```

Comment:
Maybe a library like [FluentValidation ](https://github.com/FluentValidation/FluentValidation)  would make this easier?

Reply:
Yeah, but we don't want to take up more dependencies as CB team doesn't want that.